src/flamenco/repair/fd_repair.c

```diff
@@ -525,3 +533,75 @@ fd_repair_metrics_t *
 fd_repair_get_metrics( fd_repair_t * repair ) {
   return &repair->metrics;
 }
+
+/* Pending Sign Request API
+
+   These functions manage the pool and map of pending sign requests in
+   the repair module. Each request is identified by a unique nonce,
+   allowing for nonce to be used as a key in the map.
+
+  fd_repair_pending_sign_req_t * fd_repair_acquire_pending_request(...);
+    Acquires an empty pending sign request from the pool. Returns
+    pointer or NULL if pool is full. Caller is responsible for setting
+    all fields before adding to map.
+
+  int fd_repair_add_pending_to_map(...);
+    Adds a pending sign request to the map. Returns 0 on success, -1 on
+    failure. The pending request must be previously acquired from the
+    pool.
+
+  fd_repair_pending_sign_req_t * fd_repair_find_pending_request(...);
+    Finds a pending sign request by nonce. Returns pointer or NULL.
+
+  int fd_repair_remove_pending_request(...);
+    Removes a pending sign request by nonce. Returns 0 on success, -1
+    if not found.
+
+   All functions assume the repair context is valid and not used concurrently.
+*/
+
+int
+fd_repair_pending_request_pool_free( fd_repair_t * repair ) {
+  return (int)fd_repair_pending_sign_req_pool_free( repair->pending_sign_req_pool );
+}
```

Comment:
do !free. this is unnecessary (and makes it more confusing when reading the code at usage site)